sdk/go/common/workspace/plugins_test.go

```diff
@@ -634,6 +660,33 @@ func TestPluginGetLatestVersion(t *testing.T) {
 		assert.NoError(t, err)
 		assert.Equal(t, expectedVersion, *version)
 	})
+	t.Run("Test GetLatestVersion From GitLab Releases", func(t *testing.T) {
+		t.Setenv("GITLAB_TOKEN", token)
+		spec := PluginSpec{
+			PluginDownloadURL: "gitlab://gitlab.com/278964",
+			Name:              "mock-gitlab",
+			Kind:              PluginKind("resource"),
+		}
+		expectedVersion := semver.MustParse("1.23.0")
+		source, err := spec.GetSource()
+		assert.NoError(t, err)
+		getHTTPResponse := func(req *http.Request) (io.ReadCloser, int64, error) {
+			if req.URL.String() == "https://gitlab.com/api/v4/projects/278964/releases/permalink/latest" {
```

Comment:
Minor: I think we can improve readability here with nesting and a local variable:

```go
wantURL := "https://gitlab.com/api/v4/projects/278964/releases/permalink/latest"
getHTTPResponse := func(req *http.Request) (io.ReadCloser, int64, error) {
  if req.URL.String() != wantURL {
    panic(..)
  }
  assert.Equal(t, "Bearer whatever", req.Header.Get("Authorization"))
  // ..
```

Also, you can turn that panic into a better error message with an assert instead of just an `==`:

```go
if url := req.URL.String(); !assert.Equal(t, wantURL, url) {
  panic(..)
}
```

(I would also consider dropping the panic since the test will already be marked as failed by that point.)

Reply:
Interesting idea, I'm going to leave it as is for this PR because I'd want to keep these tests consistent and it's less clear how to do that transformation for tests that will hit two URLs in succession.

Reply:
Sounds good